lib/vif-plug-providers/representor/vif-plug-representor.c

```diff
@@ -560,6 +573,106 @@ devlink_monitor_run(void)
+        VLOG_ERR("unable to initialize udev context.");
+        return;
+    }
+
+    udev_monitor = udev_monitor_new_from_netlink(udev, "kernel");
+    if (!udev_monitor) {
+        VLOG_ERR("unable to initialize udev monitor.");
+        return;
+    }
+    if (udev_monitor_filter_add_match_subsystem_devtype(
+            udev_monitor, "net", NULL) < 0) {
+        VLOG_WARN("unable to initialize udev monitor filter.");
+    }
+    if (udev_monitor_enable_receiving(udev_monitor) < 0) {
+        VLOG_ERR("unable to initialize udev monitor.");
+        return;
+    }
+    if (udev_monitor_set_receive_buffer_size(udev_monitor,
+                                             128 * 1024 * 1024) < 0) {
+        VLOG_ERR("unable to set udev receive buffer size.");
+        return;
+    }
+}
+#endif /* HAVE_UDEV */
+
+static bool
+udev_monitor_run(void)
+{
+    bool changed = false;
+#ifdef HAVE_UDEV
+    int fd;
+    char buf[IF_NAMESIZE+1];
+    size_t n_recv;
+    struct udev_device *dev;
+
+    fd = udev_monitor_get_fd(udev_monitor);
+
+    for (;;) {
+        n_recv = recv(fd, buf, 1, MSG_PEEK);
+        if (n_recv == -1) {
+            if (errno == EAGAIN) {
+                /* Nothing to do. */
+                break;
+            } else {
+                VLOG_ERR("error on udev monitor socket: %s",
+                         ovs_strerror(errno));
+            }
+        } else {
+            dev = udev_monitor_receive_device(udev_monitor);
+            if (!dev) {
+                /* Nothing to do. */
+                break;
+            }
+            const char *udev_action = udev_device_get_action(dev);
+            if (udev_action && !strcmp(udev_action, "move")) {
+                const char *ifindex_str;
+                char *cp = NULL;
+                uint32_t ifindex;
+                struct port_node *pn;
+
+                ifindex_str = udev_device_get_sysattr_value(dev, "ifindex");
+                if (!ifindex_str) {
+                    VLOG_WARN("udev: unable to get ifindex of moved netdev.");
+                    continue;
+                }
+
+                ifindex = strtol(ifindex_str, &cp, 10);
+                if (cp && cp != ifindex_str && *cp != '\0') {
+                    VLOG_WARN("udev provided malformed ifindex: '%s'",
+                              ifindex_str);
+                    continue;
+                }
+
+                pn = port_table_lookup_ifindex(port_table, ifindex);
+                if (!pn) {
+                    VLOG_DBG("udev move event on port we do not know about "
+                             "ifindex=%s", ifindex_str);
+                    continue;
+                }
+
+                if (!if_indextoname(ifindex, buf)) {
+                    VLOG_ERR("Unable to lookup netdev name from ifindex: %s",
+                             ovs_strerror(errno));
+                    continue;
+                }
+                port_node_update(pn, buf);
+                changed = true;
+            }
```

Comment:
We may need to unref the device entry we got from `udev_monitor_receive_device` here.

Reply:
Added here:
https://github.com/ovn-org/ovn-vif/pull/1/files#diff-723b13247392c98cf4f336db1fb1bfa776ceed1afaf99840535eb35b00b6b0b7R671-R672